src/controllers/user.ts

```diff
@@ -47,8 +47,11 @@ const userController = {
             res.status(statusCodes.MISSING_PARAMS).json({ status: statusCodes.MISSING_PARAMS, message: "body[platformData.codeforces.username]: Invalid or missing 'platformData.codeforces.username'" });
         } else {
             try {
-                const foundUser: IUserModel = await userDBInteractions.findByEmail(req.body.email);
-                if (foundUser) res.status(statusCodes.BAD_REQUEST).json({ status: statusCodes.BAD_REQUEST, message: "User already exists" });
+                if ( await userDBInteractions.findByEmail(req.body.email) ) {
```

Comment:
Would it make sense to move this logic from the controller to a custom validator function? For the sake of separation of concerns and organization. I think we did something similar with the Problems microservice, so it would be more consistent.

Reply:
I think my view on that is that for problems, the check was checking validity of data in the sense that it checked that the provided problem actually existed - whereas in this case the check is not that the data is valid but that it does not conflict with other users so its actual logic? Not sure if you think they're too similar though?

Reply:
There's also the fact that the same logic is used for 404s.